### PR TITLE
logger: libtcmu/qcow convert to use logger

### PR DESCRIPTION
**Pull request by lxbsz**
Signed-off-by: Xiubo Li <lixiubo@cmss.chinamobile.com>
---
These have been missed before.

### CODE REVIEW

File: qcow.c [hunk 1]
```diff
@@ -279,16 +279,29 @@ static int qcow_probe(struct bdev *bdev, int dirfd, const char *pathname)
...
 		return -1;
-	if (pread(fd, &head, sizeof(head), 0) == -1)
+	}
+	if (pread(fd, &head, sizeof(head), 0) == -1) {
+		tcmu_err("read failed fd: %d\n", fd);
 		goto err;
-	if (be32toh(head.magic) != QCOW_MAGIC)
+	}
+	if (be32toh(head.magic) != QCOW_MAGIC) {
+		tcmu_warn("not qcow: will treat as raw: %s", pathname);
```
Comment (seroyer) [inline qcow.c]: Next it will try qcow2, so you will get two of these messages for each file if it is raw.

File: qcow.c [hunk 2]
```diff
@@ -279,16 +279,29 @@ static int qcow_probe(struct bdev *bdev, int dirfd, const char *pathname)
...
 		goto err;
-	if (be32toh(head.version) != 1)
+	}
+	if (be32toh(head.version) != 1) {
+		tcmu_err("version = %d\n", head.version);
```
Comment (seroyer) [inline qcow.c]: If the file is qcow2, this will show up as an error in the log, which is misleading.

File: qcow.c (hunk 1 above)
Comment (lxbsz) [inline qcow.c]: Yes, for other errors if occurs will almost be twice too.

I will remove all logger messages in qcow later.

For the current route path the qcow/qcow2 probe will be duplicated, maybe could check the image type before probe and then just jump to it. Then we could know explicitly when and where the error triggered.

Comment (seroyer) [inline qcow.c]: Yes, that sounds like a good plan.

**Comment by mikechristie** [comment]
@seroyer, it looks ok to me. Do you have any more comments?

File: qcow.c [hunk 3]
```diff
@@ -307,15 +307,17 @@ static int qcow2_probe(struct bdev *bdev, int dirfd, const char *pathname)
...
 	}
 	if ((fd = openat(dirfd, pathname, O_RDONLY)) == -1) {
-		perror("openat failed");
+		tcmu_err("openat failed dirfd %d, pathname %s, errno %d\n",
+			 dirfd, pathname, errno);
 		return -1;
 	}
 	if (pread(fd, &head, sizeof(head), 0) == -1) {
-		perror("read fail");
+		tcmu_err("read failed fd: %d\n", fd);
```
Comment (seroyer) [inline qcow.c]: It would be more helpful if you printed the pathname and errno.  I'm not sure that the fd is overly useful here.

Comment (lxbsz) [inline qcow.c]: Has updated.

**Comment by seroyer** [comment]
Looks good to me.

**Comment by mikechristie** [comment]
Merged thanks.

**Pull request merged**